### DISCUSSION TITLE
Nonebot支持同时运行多个Bot吗

### DISCUSSION THREAD

**Discussion by ECIAP** [nonebot/discussions#46] [问答]
cqhttp应该可以通过配置多个不同端口的ws-server来完成通讯
想知道nonebot2是否可以同时运行多个bot

**Comment by yanyongyu** [comment]
不太理解你为什么需要多个server，你可以多个账号连接一个server

**Comment by mnixry** [comment]
在通信方式上, NoneBot2是支持作为服务端连接的, 即OneBot实现端(如go-cqhttp)主动连接NoneBot2, 所以NoneBot2是原生支持多个帐号连接到同一个NoneBot2实例
关于NoneBot2处理来自不同帐号消息的具体流程, 请[查看文档](https://v2.nonebot.dev/advanced/#%E5%AE%83%E5%A6%82%E4%BD%95%E5%B7%A5%E4%BD%9C)

同时, 你也可以在OneBot实现端设置多个反向WebSocket上报地址, 以实现和多个服务端同时进行通信
